pkg/apis/vpc/v1alpha1/subnetconnectionbindingmap_types.go

```diff
@@ -0,0 +1,61 @@
+package v1alpha1
+
+import metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+// +kubebuilder:validation:XValidation:rule="!(has(self.targetSubnetSetName) && has(self.targetSubnetName))",message="Only one of targetSubnetSetName or targetSubnetName can be specified"
+type SubnetConnectionBindingMapSpec struct {
+	// SubnetName is the Subnet name which this SubnetConnectionBindingMap is associated.
+	// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="subnetName is immutable"
+	SubnetName string `json:"subnetName"`
+	// TargetSubnetSetName specifies the target SubnetSet which a Subnet is bound to.
+	// +kubebuilder:validation:Optional
+	TargetSubnetSetName string `json:"targetSubnetSetName,omitempty"`
+	// TargetSubnetName specifies the target Subnet which a Subnet is bound to.
+	// +kubebuilder:validation:Optional
+	TargetSubnetName string `json:"targetSubnetName,omitempty"`
```

Comment:
do we support update targetSubnetName or targetSubnetSetName or from targetSubnetSetName to targetSubnetName in spec?

Reply:
If following NSX API design, yes, we shall support this update. @jianjuns do you feel we shall reject such modification in CR ?

Reply:
I am fine either way. If disallow update is easier to implement we can do that too.

Reply:
I don't feel it saves much code. I would keep it as it was (support user updating the parent subnet/subnetset) in the CRD and implementation.

Reply:
if targetSubnetSetName is set, the specified SubnetSet CR might do not have any NSX subnet yet. Will it report error in this case?

Reply:
In the reconciler implementation, I will add check about it, if the NSX VpcSubnet doesn't exist (mapping to the SubnetSet CR), I will add condition to describe the unready message.